include/caliper/cali_macros.h

```diff
@@ -90,6 +91,11 @@
 #define CALI_CXX_MARK_LOOP_ITERATION(loop_id, iter) \
     cali::Loop::Iteration __cali_iter_##loop_id ( __cali_loop_##loop_id.iteration(static_cast<int>(iter)) )
 
+#define CALI_TRACK_ALLOCATION(ptr) \
```

Comment:
Can we call these CALI_CXX_TRACK_ALLOCATION? Should be explicit that it's a C++-only macro